File: 132P/primes.py
#! /usr/bin/env python3
prime = [2] # prime list
n     = 3   # prime candidate
z     = 500 # number of primes upper limit
lines = 50  # number of output lines
while len (prime) < z:
  prime.append (n)
  while True:
    n += 2
    try:
      for p in prime [1:]:
        if n % p == 0: break              # nah! ck next
        if n// p <= p: raise Exception () # yup!
    except: break
for i in range (lines):
  buf = ''
  for j in range (i, z, lines):
    buf += ' ' + str (prime [j]).rjust (4, '0')
  print (buf)
################################################################
# log:

File: 132P/primes.pl
#!/usr/bin/env perl
use strict;
use warnings;

my @prime = (2,3);

sub isprime {
  my ($n) = @_;
  for (my $k = 1;; $k++) {
    my $p = $prime[$k];
    if ($n%$p ==  0) { return 0; }
    if ($n/$p <= $p) { return 1; }
  }
}
sub gen_primes {
  my ($size) = @_;
  for (my $n = 5; $#prime < $size; $n += 2) {
    if (isprime($n)) {
      push @prime, $n;
    }
  }
}
sub spit {
  my ($linez) = @_;
  my $size = $#prime;
  for (my $i = 0; $i < $linez; ++$i) {
    my $s = "";
    for (my $j = $i; $j < $size; $j += $linez) {
      $s = $s . sprintf(" %04d", $prime[$j]);
    }
    print "$s\n";
  }
}
gen_primes(500);
spit(50);
#log:
